### PR TITLE
feat: add standalone extractor for textproto

### PR DESCRIPTION
**Pull request by justbuchanan**
The textproto extractor looks for `# proto-file:` and `# proto-message:` comments OR flags to determine what the schema of the textproto file is. The proto extractor then does most of the heavy lifting of tracking down and bundling up all of the proto dependencies.

The tests are using the same kzip diffing script/bzl as the proto extractor tests.

### CODE REVIEW

File: kythe/cxx/extractor/textproto/textproto_extractor.cc [hunk 1]
```diff
@@ -0,0 +1,197 @@
...
+
+/// \brief Returns a kzip-based IndexWriter or dies.
+IndexWriter OpenKzipWriterOrDie(absl::string_view path) {
+  auto writer = KzipWriter::Create(path);
+  CHECK(writer.ok()) << "Failed to open KzipWriter: " << writer.status();
+  return std::move(*writer);
+}
+
+// Loads all data from a file or terminates the process.
+static std::string LoadFileOrDie(const std::string& file) {
```
Comment (shahms) [inline kythe/cxx/extractor/textproto/textproto_extractor.cc]: `static` is redundant with anonymous namespace

File: kythe/cxx/extractor/textproto/textproto_extractor.cc [hunk 2]
```diff
@@ -0,0 +1,197 @@
...
+  FILE* handle = fopen(file.c_str(), "rb");
+  CHECK(handle != nullptr) << "Couldn't open input file " << file;
+  CHECK_EQ(fseek(handle, 0, SEEK_END), 0) << "Couldn't seek " << file;
+  long size = ftell(handle);
+  CHECK_GE(size, 0) << "Bad size for " << file;
+  CHECK_EQ(fseek(handle, 0, SEEK_SET), 0) << "Couldn't seek " << file;
+  std::string content;
+  content.resize(size);
+  CHECK_EQ(fread(&content[0], size, 1, handle), 1) << "Couldn't read " << file;
```
Comment (shahms) [inline kythe/cxx/extractor/textproto/textproto_extractor.cc]: content.data()

File: kythe/cxx/extractor/textproto/textproto_extractor.cc [hunk 3]
```diff
@@ -0,0 +1,197 @@
...
+  // the textproto itself or as command line flags to the extractor. Note that
+  // if metadata is specified both in the textproto and via flags, flags take
+  // precedence.
+  TextprotoSchema schema = ParseTextprotoSchemaComments(textproto);
+  std::vector<std::string> proto_filenames;
+  if (!FLAGS_proto_files.empty()) {
+    proto_filenames = absl::StrSplit(FLAGS_proto_files, ",");
+  } else {
+    proto_filenames.push_back(schema.proto_file);
+    for (auto& extra_import : schema.proto_imports) {
```
Comment (shahms) [inline kythe/cxx/extractor/textproto/textproto_extractor.cc]: `const auto&`

File: kythe/cxx/extractor/textproto/textproto_extractor.cc (hunk 2 above)
Comment (justbuchanan) [reply kythe/cxx/extractor/textproto/textproto_extractor.cc]: The `const`ness of `.data()` gets in the way here since we're writing to it.

Comment (shahms) [reply kythe/cxx/extractor/textproto/textproto_extractor.cc]: Forgot non-const `data()` wasn't added until C++17

**Pull request merged**